DOC-413 Specify ports for firewall vs open ports on systems (#1205)
RedisLabs/redislabs-docs: content/rs/administering/designing-production/networking/port-configurations.md

| TCP | 8443 | Internal, External | Secure (HTTPS) access to the management web UI |
| TCP | 9081 | Internal, Active-Active | Active-Active management |
| TCP | 9443 (Recommended), [8080](#turning-off-http-support) | Internal, External, Active-Active | REST API traffic, including cluster management and node bootstrap |
| TCP | 10000-19999 | Internal, External, Active-Active | Database traffic |
| UDP | 53, 5353 | Internal, External | DNS/mDNS traffic |
| ICMP | * | Internal | Connectivity checking between nodes |
| TCP | 1968 | Internal | Proxy traffic |
| TCP | 3333-3341, 3343-3344, 36379, 36380 | Internal | Internode communication |
| TCP | 20000-29999 | Internal | Database shard traffic |
| TCP | 8002, 8004, 8006 | Internal | System health monitoring |
| TCP | 8444, 9080 | Internal | Traffic between web proxy and cnm_http/cm |

## Change the admin console port

The Redis Enterprise Software admin console uses port 8443, by default. You can change this to a custom port as long as the new port is not in use by another process.

To change this port, run:

```sh
rladmin cluster config cm_port <new-port>
```

> After changing the Redis Enterprise Software web UI port, you must connect any new node added to the cluster to the UI with the custom port number:

Reviewer: ```suggestion
New nodes added the cluster will not default to the new admin console port number. You must manually connect the new nodes to the admin console: 
```

How about we reword to this? @banker @AlonMagrafta

Reviewer 2: @kaitlynmichael This is the sentence that I don't understanding the meaning of. What does it mean to manually connect new nodes to the admin console?

alwaysopen: false
categories: ["RS"]
---

> To make sure Redis Enterprise Software servers can pass necessary communications between them,

Reviewer 2: ```suggestion
All Redis Enterprise Software deployments span multiple physical/virtual nodes. You'll need to keep several ports open between these nodes. This document describes the various port ranges and their uses.
```

> we recommend all servers have the following ports open.

Reviewer 2: ```suggestion
```

{{% Note %}}

> You must manually update your firewall with the port for each new database endpoint. The cluster assigns ports (between 10,000 and 19,999) for database endpoints by default, and only verifies that new endpoints are not already in use.

Reviewer 2: ```suggestion
Whenever you create a new database, you must verify that the ports assigned to the new database's endpoints are open. The cluster will not perform this verification for you. 
```

{{% /Note %}}

## Ports and port ranges used by Redis Enterprise Software

> There are 3 types of ports that are open for use by RS:

Reviewer 2: ```suggestion
Redis Enterprise Software's port usage falls into three general categories:
```

> - Internal - The traffic is between or within the cluster nodes

Reviewer 2: ```suggestion
- Internal: For traffic between or within cluster nodes
```

> - External - The traffic is from client applications or external monitoring resources

Reviewer 2: ```suggestion
- External: For traffic from client applications or external monitoring resources
```

> - Active-Active - The traffic is from clusters that host Active-Active databases

Reviewer 2: ```suggestion
- Active-Active: For traffic to and from clusters hosting Active-Active databases
```

> ### Ports for traffic between the cluster and other systems

Reviewer 2: @kaitlynmichael I find it really confusing that we describe Internal, External, and Active-Active ports but then don't stick with that organization scheme as we describe the ports below.

Reviewer: Agreed. I didn't consider that in my first pass.

Reviewer: I combined the table into one since the internal/external/active-active distinction is already made within the table. We don't need the extra headings

| TCP | 9443 (Recommended), [8080](#turning-off-http-support) | Internal, External, Active-Active | REST API traffic, including cluster management and node bootstrap |
| TCP | 10000-19999 | Internal, External, Active-Active | Database traffic |
| UDP | 53, 5353 | Internal, External | DNS/mDNS traffic |

### Ports for traffic between cluster nodes

These ports are used by Redis Software between cluster nodes:

| Protocol | Port | Connection Source | Description |
|------------|-----------------|-----------------|-----------------|
| ICMP | * | Internal | Connectivity checking between nodes |
| TCP | 1968 | Internal | Proxy traffic |
| TCP | 3333-3341, 3343-3344, 36379, 36380 | Internal | Internode communication |
| TCP | 20000-29999 | Internal | Database shard traffic |

### Ports for traffic within the cluster nodes

These ports are used by Redis Software within each cluster node:

| Protocol | Port | Connection Source | Description |
|------------|-----------------|-----------------|-----------------|
| TCP | 8002, 8004, 8006 | Internal | System health monitoring |
| TCP | 8444, 9080 | Internal | Traffic betwen web proxy and cnm_http/cm |

> ## Changing the management web UI port

Reviewer 2: ```suggestion
## Changing the admin console port
```

> You can change the default port for the Redis Enterprise Software web UI (8443) to a custom port, as long as the new port is not in use by another process.

Reviewer 2: ```suggestion
The Redis Enterprise Software admin console uses port 8443, by default. You can change this to a custom port as long as the new port is not in use by another process.
```

> To change the default port for the RS Web UI, on any node in the cluster run:

Reviewer 2: ```suggestion
To change this port, run:
```

```sh
rladmin cluster config cm_port <new-port>
```

After changing the Redis Enterprise Software web UI port, you must connect any new node added to the cluster to the UI with the custom port number:
`https://newnode.mycluster.example.com:`**`<nonstandard-port-number>`**

> 

Reviewer 2: @kaitlynmichael This section strikes me as unnecessary and unclear. Is it saying that you need to connect over the web using the new port? That should be obvious, I think. Or does it mean that you'll use this port when adding a new node? It's the example here that makes this super unclear. cc @AlonMagrafta

Reviewer: We will have to address this in a separate PR as this needs to be updated ASAP

> ## Disabling HTTP support for API endpoints

Reviewer 2: This belongs in the Security section, not here. I've created a ticket to address this.

that the assigned port is not already in use. You must manually
update your firewall with the port for that new database endpoint.

## Ports and port ranges used by RS

There are 3 types of ports that are open for use by RS:

- Internal - The traffic is between or within the cluster nodes
- External - The traffic is from client applications or external monitoring resources
- Active-Active - The traffic is from clusters that host Active-Active databases

### Network traffic between the cluster and other systems

You need to open these ports in your firewall to allow traffic to pass from the cluster to other systems:

| Protocol | Port | Connection Source | Description |
|------------|-----------------|-----------------|-----------------|
| TCP | 8001 | Internal, External | Traffic from application to RS [Discovery Service]({{< relref "/rs/concepts/data-access/discovery-service.md" >}}) |
| TCP | 8070, 8071 | Internal, External | For metrics exported and managed by the web proxy |
| TCP | 8443 | Internal, External | Secure (HTTPS) access to the management web UI |
| TCP | 9081 | Internal, Active-Active | For Active-Active management |

> | TCP | 9443 (Recommended), [8070, 8080](#turning-off-http-support) | Internal, External, Active-Active | REST API traffic, including cluster management and node bootstrap |

Reviewer 3: ```suggestion
| TCP | 9443 (Recommended), [8080](#turning-off-http-support) | Internal, External, Active-Active | REST API traffic, including cluster management and node bootstrap |
```
